File: stepik-java-api/src/main/java/org/stepik/api/queries/attempts/StepikAttemptsGetQuery.java
package org.stepik.api.queries.attempts;

import org.jetbrains.annotations.NotNull;
import org.stepik.api.actions.StepikAbstractAction;
import org.stepik.api.objects.attempts.Attempts;
import org.stepik.api.queries.StepikAbstractGetQuery;
import org.stepik.api.urls.Urls;

/**
 * @author meanmail
 */
public class StepikAttemptsGetQuery extends StepikAbstractGetQuery<StepikAttemptsGetQuery, Attempts> {
    public StepikAttemptsGetQuery(@NotNull StepikAbstractAction stepikAction) {
        super(stepikAction, Attempts.class);
    }

    public StepikAttemptsGetQuery step(long id) {
        addParam("step", id);
        return this;
    }

    @NotNull
    @Override
    protected String getUrl() {
        return Urls.ATTEMPTS;
    }
}

File: stepik-java-api/src/main/kotlin/org/stepik/api/queries/attempts/StepikAttemptsGetQuery.kt
package org.stepik.api.queries.attempts

import org.stepik.api.actions.StepikAbstractAction
import org.stepik.api.objects.attempts.Attempts
import org.stepik.api.queries.StepikAbstractGetQuery

class StepikAttemptsGetQuery(stepikAction: StepikAbstractAction) :
        StepikAbstractGetQuery<StepikAttemptsGetQuery, Attempts>(stepikAction, Attempts::class.java) {
    
    fun step(id: Long): StepikAttemptsGetQuery {
        addParam("step", id)
        return this
    }
    
    fun user(userId: Long): StepikAttemptsGetQuery {
        addParam("user", userId)
        return this
    }
    
    override val url = "${stepikAction.stepikApiClient.host}/api/attempts"
    
}
